rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/queue/RedisCircularBufferDAO.java

```diff
@@ -0,0 +1,22 @@
+package com.lithium.dbi.rdbi.recipes.queue;
+
+import com.lithium.dbi.rdbi.BindArg;
+import com.lithium.dbi.rdbi.BindKey;
+import com.lithium.dbi.rdbi.Query;
+
+public interface RedisCircularBufferDAO {
+
+    @Query(
+            "local size = redis.call('RPUSH', $key$, $valueToAdd$)\n" +
+                    "if size > tonumber($maxSize$) then\n" +
```

Comment:
what if something has gone wrong and the current size is way larger than max size?